docs/src/main/asciidoc/reaugmentation.adoc

```diff
@@ -0,0 +1,66 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Reaugment a Quarkus Application
+
+include::./attributes.adoc[]
+
+== What is augmentation?
+
+During augmentation the application is initialized - or let's say pre-initialized.
```

Comment:
This may sound more complicated but also perhaps more accurate?

Quarkus application configuration may include two types of configuration options: build time options, that are handled during the application build time, and runtime options that may be adjusted after the application has been build but before it's been launched.
The augmentation is a phase of an application build process during which the byte code of the application is optimized according to the application build time configuration.